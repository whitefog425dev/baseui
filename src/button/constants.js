/* eslint-disable flowtype/require-valid-file-annotation */

export const KIND = {
  primary: 'primary',
  secondary: 'secondary',
  tertiary: 'tertiary',
  minimal: 'minimal',
};

export const SHAPE = {
  default: 'default',
  round: 'round',
  square: 'square',
};

export const SIZE = {
  default: 'default',
  compact: 'compact',
};
