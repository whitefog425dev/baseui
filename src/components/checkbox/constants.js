// @flow
/* eslint-disable import/prefer-default-export */

export const STATE_TYPE = {
  change: 'CHANGE',
};

/* eslint-enable import/prefer-default-export */
