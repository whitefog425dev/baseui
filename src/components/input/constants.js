// @flow
export const STATE_CHANGE_TYPE = {
  change: 'change',
};

export const ADJOINED = {
  none: 'none',
  left: 'left',
  right: 'right',
  both: 'both',
};

export const SIZE = {
  default: 'default',
  compact: 'compact',
};

export const ENHANCER_POSITION = {
  start: 'start',
  end: 'end',
};
