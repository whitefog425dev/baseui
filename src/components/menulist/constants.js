// @flow
/* eslint-disable import/prefer-default-export */

export const STATE_CHANGE_TYPES = {
  moveUp: 'moveUp',
  moveDown: 'moveDown',
};

// Dict of props to prepend $ to pass thru to styletron
export const STYLETRON_PROP_MAPPER = {
  ref: true,
  isHighlighted: true,
};

export const KEY_STRINGS = {
  ArrowUp: 'ArrowUp',
  ArrowDown: 'ArrowDown',
  Enter: 'Enter',
};
