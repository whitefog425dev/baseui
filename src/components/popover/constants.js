/*
MIT License

Copyright (c) 2018 Uber Technologies, Inc.

Permission is hereby granted, free of charge, to any person obtaining a copy
of this software and associated documentation files (the "Software"), to deal
in the Software without restriction, including without limitation the rights
to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
copies of the Software, and to permit persons to whom the Software is
furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in
all copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
THE SOFTWARE.
*/
// @flow

export const PLACEMENT = {
  auto: 'auto',
  topLeft: 'topLeft',
  top: 'top',
  topRight: 'topRight',
  rightTop: 'rightTop',
  right: 'right',
  rightBottom: 'rightBottom',
  bottomRight: 'bottomRight',
  bottom: 'bottom',
  bottomLeft: 'bottomLeft',
  leftBottom: 'leftBottom',
  left: 'left',
  leftTop: 'leftTop',
};

export const TRIGGER_TYPE = {
  click: 'click',
  hover: 'hover',
};

export const STATE_CHANGE_TYPE = {
  open: 'open',
  close: 'close',
};

export const ACCESSIBILITY_TYPE = {
  none: 'none',
  menu: 'menu',
  tooltip: 'tooltip',
};

export const POPOVER_MARGIN = 8;

export const ARROW_SIZE = 6;

/**
 * Since we use a 45-degree rotated div to render the arrow, the
 * width/height of this div is different than the arrow size itself
 *
 * The arrow size is essentially half the diagonal of the rotated div,
 * using pythagorean theorem:
 *   width^2 + height^2 = (arrow_size * 2)^2
 * In this case width = height so:
 *   2 * width^2 = (arrow_size * 2)^2
 * Simplifies to:
 *   width = âˆš((arrow_size * 2)^2 / 2)
 */
export const ARROW_WIDTH = Math.ceil(Math.sqrt((ARROW_SIZE * 2) ** 2 / 2));
