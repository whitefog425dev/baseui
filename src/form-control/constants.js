/*
Copyright (c) 2018 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/
// @flow
// Dictionary of props to prepend $ to pass through to styletron
export const STYLETRON_PROP_MAPPER = {
  disabled: true,
  error: true,
  required: true,
  size: true,
};
