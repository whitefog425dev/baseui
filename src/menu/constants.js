/*
Copyright (c) 2018 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/
// @flow

export const STATE_CHANGE_TYPES = {
  moveUp: 'moveUp',
  moveDown: 'moveDown',
};

// Dict of props to prepend $ to pass thru to styletron
export const SHARED_PROPS_MAPPER = {
  ref: true,
  isHighlighted: true,
};

export const KEY_STRINGS = {
  ArrowUp: 'ArrowUp',
  ArrowDown: 'ArrowDown',
  Enter: 'Enter',
  Space: ' ',
  Escape: 'Escape',
  Backspace: 'Backspace',
};

export const OPTION_LIST_SIZE = {
  default: 'default',
  compact: 'compact',
};
