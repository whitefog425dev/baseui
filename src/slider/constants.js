// @flow

export const STATE_CHANGE_TYPE = {
  change: 'change',
};

export const thumbWidth = 32;
