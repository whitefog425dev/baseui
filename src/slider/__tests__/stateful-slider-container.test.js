/*
Copyright (c) 2018 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/
// @flow
import React from 'react';
import {mount} from 'enzyme';
import {StatefulContainer as StatefulSliderContainer} from '../index';
import {STATE_CHANGE_TYPE} from '../constants';
import type {StateReducerT} from '../types';

describe('Stateful Slider Container', function() {
  let allProps: any, childFn;
  let wrapper;

  beforeEach(function() {
    const stateReducer: StateReducerT = (type, nextState) => nextState;
    childFn = jest.fn(() => <div>test</div>);
    allProps = {
      children: childFn,
      initialState: {},
      stateReducer: stateReducer,
      prop1: 'some other slider propq',
    };
  });

  afterEach(function() {
    jest.restoreAllMocks();
    wrapper && wrapper.unmount();
  });

  test('should correctly render', function() {
    wrapper = mount(<StatefulSliderContainer {...allProps} />);
    expect(wrapper).toMatchSnapshot('Component has correct render ');
  });

  test('should provide all needed props to children render func', function() {
    wrapper = mount(<StatefulSliderContainer {...allProps} />);
    const actualProps = childFn.mock.calls[0][0];
    expect(actualProps).toMatchObject({
      prop1: allProps.prop1,
    });
  });

  test('should provide initial state as part of state', function() {
    allProps.initialState = {value: [30, 80]};
    wrapper = mount(<StatefulSliderContainer {...allProps} />);
    const actualProps = childFn.mock.calls[0][0];
    expect(actualProps).toMatchObject(allProps.initialState);
  });

  describe('Events', function() {
    let value = [40];
    let events, stateReducerMock, instance, event;
    event = {target: {}};
    const handlers = [['onChange', STATE_CHANGE_TYPE.change, {value}]];
    beforeEach(function() {
      events = {
        onChange: jest.fn(),
      };
      allProps = {...allProps, ...events};
      stateReducerMock = jest.fn();
      allProps.stateReducer = stateReducerMock;
      wrapper = mount(<StatefulSliderContainer {...allProps} />);
      instance = wrapper.instance();
    });

    test.each(handlers)(
      'should call state reducer to apply new state for %s event with %s type',
      (eventHandler, type, newState) => {
        const handler = instance[eventHandler];
        const params = {...newState};
        handler({event, ...params});
        expect(stateReducerMock).toHaveBeenCalledWith(
          type,
          newState,
          {},
          event,
        );
        expect(events[eventHandler]).toHaveBeenCalledWith({event, ...params});
      },
    );
  });
});
