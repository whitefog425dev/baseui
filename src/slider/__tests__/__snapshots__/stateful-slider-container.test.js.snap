// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stateful Slider Container should correctly render: Component has correct render  1`] = `
<StatefulSliderContainer
  initialState={Object {}}
  onChange={[Function]}
  prop1="some other slider propq"
  stateReducer={[Function]}
>
  <div>
    test
  </div>
</StatefulSliderContainer>
`;
